src/main/java/com/codeborne/selenide/impl/CollectionElement.java

```diff
@@ -3,30 +3,42 @@
 import com.codeborne.selenide.Condition;
 import com.codeborne.selenide.SelenideElement;
 import com.codeborne.selenide.ex.ElementNotFound;
+import org.openqa.selenium.StaleElementReferenceException;
 import org.openqa.selenium.WebElement;
 
 import java.lang.reflect.Proxy;
+import java.util.List;
 
 import static com.codeborne.selenide.Condition.visible;
 
 public class CollectionElement extends WebElementSource {
-  public static SelenideElement wrap(WebElementsCollection collection, int index) {
+  public static SelenideElement wrap(WebElementsCollection collection, List<WebElement> actualElements, int index) {
     return (SelenideElement) Proxy.newProxyInstance(
         collection.getClass().getClassLoader(), new Class<?>[]{SelenideElement.class},
-        new SelenideElementProxy(new CollectionElement(collection, index)));
+        new SelenideElementProxy(new CollectionElement(collection, actualElements, index)));
   }
 
   private final WebElementsCollection collection;
+  private List<WebElement> actualElements;
   private final int index;
 
-  CollectionElement(WebElementsCollection collection, int index) {
+  CollectionElement(WebElementsCollection collection, List<WebElement> actualElements, int index) {
     this.collection = collection;
+    this.actualElements = actualElements;
     this.index = index;
   }
 
   @Override
   public WebElement getWebElement() {
-    return collection.getActualElements().get(index);
+    try {
+      WebElement el = actualElements.get(index);
+      el.isEnabled(); // check staleness
```

Comment:
It makes iteration slower. Every call to `el.isEnabled()` takes some time (because it sends an http request from webdriver to the browser).

But anyway, this solution is better than it was before. I think we can merge this PR and then think how to optimize this method further.

Reply:
You're right. It makes it a bit slower. But I couldn't find any other ways to check element staleness  without sending a request to browser. 😞